client/service/src/config.rs

```diff
@@ -28,6 +28,27 @@ use sp_core::crypto::Protected;
 use target_info::Target;
 use sc_telemetry::TelemetryEndpoints;
 
+/// Executable version. Used to pass version information from the root crate.
+#[derive(Clone)]
+pub struct VersionInfo {
+	/// Implementation name.
+	pub name: &'static str,
+	/// Implementation version.
+	pub version: &'static str,
+	/// SCM Commit hash.
+	pub commit: &'static str,
+	/// Executable file name.
+	pub executable_name: &'static str,
+	/// Executable file description.
+	pub description: &'static str,
+	/// Executable file author.
+	pub author: &'static str,
+	/// Support URL.
+	pub support_url: &'static str,
+	/// Copyright starting year (x-current year)
+	pub copyright_start_year: i32,
+}
+
```

Comment:
This is moved here because I need it in the `Configuration::new()` and actually this is fine. Part of the logic that determine the version, OS, arch... was already here. To keep the original API I re-export `VersionInfo` in `sc_cli` too,